src/js/conduit/Conduit.js

```diff
@@ -223,346 +315,499 @@ export default class Conduit {
-   *   Resolves the streamName when the response is received AND is successful.
-   *   Rejects if the transaction fails or if the response code is not 200.
+   * @async
+   *
+   * @returns {String}
+   *   the stream name
    */
-  validateHash () {
+  async validateHash () {
     this.logger.debug('Validating Hash...');
 
-    return new Promise((resolve, reject) => {
-      try {
-        this.transaction(
-          'iov/hashValidate',
-          {
-            b64HashURL: this.b64_hash_access_url,
-            token: this.hash,
-          },
-          (response) => {
-            // response ->  {"status": 200, "target_url": "clsp://sfs1/fakestream", "error": null}
-
-            if (response.status !== 200) {
-              if (response.status === 403) {
-                return reject(new Error('HashUnAuthorized'));
-              }
-
-              return reject(new Error('HashInvalid'));
-            }
-
-            //TODO, figure out how to handle a change in the sfs url from the
-            // clsp-hash from the target url returned from decrypting the hash
-            // token.
-            // Example:
-            //    user enters 'clsp-hash://sfs1/hash?start=0&end=...&token=...' for source
-            //    clspUrl = 'clsp://SFS2/streamOnDifferentSfs
-            // --- due to the videojs architecture i don't see a clean way of doing this.
-            // ==============================================================================
-            //    The only way I can see doing this cleanly is to change videojs itself to
-            //    allow the 'canHandleSource' function in MqttSourceHandler to return a
-            //    promise not a value, then ascychronously find out if it can play this
-            //    source after making the call to decrypt the hash token.22
-            // =============================================================================
-            // Note: this could go away in architecture 2.0 if MQTT was a cluster in this
-            // case what is now the sfs ip address in clsp url will always be the same it will
-            // be the public ip of cluster gateway.
-            const t = response.target_url.split('/');
-
-            // get the actual stream name
-            const streamName = t[t.length - 1];
-
-            resolve(streamName);
-          }
-        );
-      }
-      catch (error) {
-        reject(error);
-      }
+    // response ->  {"status": 200, "target_url": "clsp://sfs1/fakestream", "error": null}
+    const {
+      payloadString: response,
+    } = await this.transaction('iov/hashValidate', {
+      b64HashURL: this.streamConfiguration.b64_hash_access_url,
+      token: this.streamConfiguration.hash,
     });
+
+    if (response.status === 403) {
+      throw new Error('HashUnAuthorized');
+    }
+
+    if (response.status !== 200) {
+      throw new Error('HashInvalid');
+    }
+
+    // TODO, figure out how to handle a change in the sfs url from the
```

Comment:
i didn't contribute this portion of the code, so I'm not comfortable removing or consolidating - I'll talk to the author because I think one of those functions can be removed entirely

Reply:
i have completely removed the references to the jwt functionality.  this comment is no longer duplicated.  is it still ok in it's current state?